File: Data Structures & Algorithms/remove-element/submission-2.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        i = 0
        while i < len(nums):
            while i < len(nums) and nums[i] != val:
                i += 1
            j = i+1
            while j < len(nums) and nums[j] == val:
                j += 1
            if j >= len(nums):
                return i
            nums[i], nums[j] = nums[j], nums[i]
        return len(nums)
            
        
            

File: Data Structures & Algorithms/two-integer-sum/submission-2.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> map = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++) {
            int n = target-nums[i];
            if (map.ContainsKey(n)){
                return new int[2]{map[n], i};
            }
            map[nums[i]] = i;
        }
        return new int[2]{0,1};
    }
}
